src/dvclive/live.py

```diff
@@ -263,7 +265,18 @@ def metrics_file(self) -> str:
 
     @property
     def dvc_file(self) -> str:
-        return os.path.join(self.dir, "dvc.yaml")
+        if isinstance(self._dvcyaml, str):
+            if os.path.basename(self._dvcyaml) == "dvc.yaml":
+                return self._dvcyaml
+            raise InvalidDvcyamlError
+        if self._dvc_repo is not None:
+            return os.path.join(self._dvc_repo.root_dir, "dvc.yaml")
+        self._dvcyaml = False
+        logger.warning(
+            "Can't infer dvcyaml path without a DVC repo. "
+            "`dvc.yaml` file will not be written."
+        )
+        return ""
```

Comment:
Can we use https://docs.python.org/3/library/functools.html#functools.cached_property?

Reply:
Okay, moved it inside a new `_init_dvc_file` method since `_init_dvc` is already convoluted. PTAL. Not sure if there's value in keeping the `dvc_file` property method here vs. setting `self.dvc_file` directly in `_init_dvc`, but no strong opinion.

Reply:
> Not sure if there's value in keeping the dvc_file property method here vs. setting self.dvc_file directly in _init_dvc, but no strong opinion.

I think the value is that users can't modify/set the value of `dvc_file` directly, which is what we want